openedx/core/djangoapps/course_groups/partition_generator.py

```diff
@@ -0,0 +1,72 @@
+"""
+The team dynamic partition generation to be part of the
+openedx.dynamic_partition plugin.
+"""
+import logging
+
+from django.utils.translation import gettext_lazy as _
+from openedx.core.djangoapps.course_groups.flags import CONTENT_GROUPS_FOR_TEAMS
+
+from xmodule.partitions.partitions import UserPartition, UserPartitionError
+from xmodule.services import TeamsConfigurationService
+
+log = logging.getLogger(__name__)
+
+MINIMUM_DYNAMIC_TEAM_PARTITION_ID = 51
+TEAM_SCHEME = "team"
+
+
+def create_team_set_partition_with_course_id(course_id, team_sets=None):
+    """
+    Create and return the dynamic enrollment track user partition based only on course_id.
+    If it cannot be created, None is returned.
+    """
+    if not team_sets:
+        team_sets = get_team_sets(course_id) or {}
+
+    try:
+        team_scheme = UserPartition.get_scheme(TEAM_SCHEME)
+    except UserPartitionError:
+        log.warning("No 'team' scheme registered, TeamUserPartition will not be created.")
+        return None
+
+    # Get team-sets from course and create user partitions for each team-set
+    # Then get teams from each team-set and create user groups for each team
+    partitions = []
+    for team_set in team_sets:
+        partition = team_scheme.create_user_partition(
+            id=team_set.dynamic_user_partition_id,
+            name=f"Team set {team_set.name} groups",
```

Comment:
In https://openedx.atlassian.net/wiki/spaces/COMM/pages/3885662226/Connecting+Teams+to+Content+Groups?focusedCommentId=4027809880 I have proposed changing this text to `Team Grouping: {team_set.name}`. I will follow up in #wg-product to make sure that the conversation moves forward quickly.

Regardless of the outcome of that, @mariajgrimaldi I think this string needs to be internationalized?

Reply:
Here are my two cents on the naming issue: 

The enrollment tracks partition heavily inspires this implementation, so we used the same naming standard as there:

[name=_("Enrollment Track Groups"),](https://github.com/openedx/edx-platform/blob/master/xmodule/partitions/enrollment_track_partition_generator.py#L22-L42)

What's listed here is a [Group](https://github.com/openedx/edx-platform/blob/master/xmodule/partitions/partitions.py#L51-L55) that belongs to the partition being used, which could be an enrollment track or a team set.